yapapi/rest/market.py

```diff
@@ -112,11 +124,28 @@ async def terminate(self, reason: dict) -> bool:
         try:
             await self._api.terminate_agreement(self._id, request_body=reason)
             logger.debug("terminateAgreement(%s) returned successfully", self._id)
+            self._state.terminate()
             return True
         except (ApiException, asyncio.TimeoutError, aiohttp.ClientOSError):
             logger.debug("terminateAgreement(%s) failed", self._id, exc_info=True)
             return False
 
+    @property
+    def confirmed(self):
+        return self._state.current_state == AgreementState.confirmed
+
+    @property
+    def terminated(self):
+        return self._state.current_state == AgreementState.terminated
+
+    async def get_provider_property(self, prop_key):
+        """Retrieve a value of a provider-side (Offer) property of the Agreement."""
+        return (await self.details()).provider_view.properties.get(prop_key)
```

Comment:
well, @johny-b because: `Method cached_details() can be called only after prior `await details()`

and calls to the API are async...

Reply:
OK, but `await details()` is called **always** when the agreement is created, and we already use `cached_details` in some places assuming exactly this.

Calls to the API are async, OK, but - considering we always have details already downloaded - there is no need for an API call here.

Reply:
again, it's called _now_ but it won't be guaranteed if we change _that_ part of the code ... I'd rather have a stable interface that doesn't depend on random other places in the code

Reply:
OK, I understand your point, but we already use this assumption in many places.

For me, this is a tradeoff between:
A) Have a little less refactoring **if** we one day decide to change a lot of code (but to preserve the Agreement as it is now)
B) Have **now** a library that is easier to comprehend, because similar things are done in a similar way. And also a little more efficient.

IMHO, considering current state of `yapapi` and it's future perspectives, B is obviously more important.

Reply:
as discussed